cmd/src/debug_comp.go

```diff
@@ -0,0 +1,228 @@
+package main
+
+import (
+	"archive/zip"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+
+	"golang.org/x/sync/errgroup"
+
+	"golang.org/x/sync/semaphore"
+)
+
+func init() {
+	usage := `
+'src debug compose' invokes docker cli diagnostic commands targeting a containers that are members of a docker-compose network, 
+writing an archive file from their returns. 
+
+Usage:
+
+	src debug compose [command options]
+
+Flags:
+
+	-o					Specify the name of the output zip archive.
+	--no-configs		Don't include Sourcegraph configuration json.
+
```

Comment:
Rather than trying to get the indentation on the flags right (which is super annoying), you might want to reuse the pattern we use in other commands where we invoke `flagSet.PrintDefaults()` to document the command flags. In that case, you could drop the `Flags` section out here and instead add a call to `flagSet.PrintDefaults()` that looks something like [this example from `src api`](https://sourcegraph.com/gitlab.com/sourcegraph/src-cli@c5a7ae0c5e82e439049568a0db674fc2a52d6ba6/-/blob/cmd/src/api.go?L46-50):

```go
	usageFunc := func() {
		fmt.Fprintf(flag.CommandLine.Output(), "Usage of 'src %s':\n", flagSet.Name())
		flagSet.PrintDefaults()
		fmt.Println(usage)
	}
```

Reply:
cc @DaedalusG Might be good to follow up on this one as well.

cc @bobheadxi Any plans for or recent updates to src cli flag standards?

Reply:
`sg` delegates documentation formatting concerns to the underlying library, `ufrave/cli`, to avoid issues like this. I haven't looked at how `src` does things and it's not currently on our plans to work on `src` though

Reply:
Ok let's embed sg/src-cli into sg/sg so robert can own it